zokrates_stdlib/stdlib/hashes/mimcSponge/mimcSponge.zok

```diff
@@ -1,22 +1,19 @@
 import "./mimcFeistel" as MiMCFeistel
 
-def main(field[2] ins, field k) -> field[3]:
-	//nInputs = 2, nOutputs = 3,
-	u32 nInputs = 2
-	u32 nOutputs = 3 
-	field[4][2] S = [[0; 2]; 4] // Dim: (nInputs + nOutputs - 1, 2)
-	field[3] outs = [0; 3]
+def main<N, P>(field[N] ins, field k) -> field[P]:
```

Comment:
made this generic

Reply:
Wow this looks pretty cool. Can you think of cases where this would be a breaking change?

Reply:
Hmm good question, only https://github.com/Zokrates/ZoKrates/issues/791 comes to mind

Reply:
Okay good then, can we have a comment above this function explaining what these parameters are? Maybe linking to a spec?